presto-matching/src/test/java/com/facebook/presto/matching/TestMatcher.java

```diff
@@ -70,11 +70,13 @@ public void propertyMatchers()
         Property<String, Integer> length = Property.property("length", String::length);
         String string = "a";
 
+        assertMatch(aString.with(length.equalTo(1)), string);
```

Comment:
So far you could use:
```
assertMatch(aString.with(length.matching(x -> x.equals(1)), string);
```

There is a value in DSL like this, but I do not see much value in specialized `Pattern` I would just reuse `FilterPattern`. Is it because pattern printer?